suite-common/wallet-utils/src/localizeNumber.ts

```diff
@@ -1,28 +1,39 @@
+import BigNumber from 'bignumber.js';
+
 export const localizeNumber = (
-    amount: number,
+    value: number | string | BigNumber,
     locale = 'en',
     minDecimals = 0,
     maxDecimals = 20,
 ): string => {
-    if (
-        typeof amount !== 'number' ||
-        Number.isNaN(amount) ||
-        !Number.isFinite(amount) ||
-        amount > Number.MAX_SAFE_INTEGER ||
-        amount < Number.MIN_SAFE_INTEGER
-    ) {
-        return '';
-    }
-
     if (maxDecimals < minDecimals) {
         throw Error(
             `maxDecimals (${maxDecimals}) cannot be lower than minDecimals (${minDecimals})`,
         );
     }
 
-    return Intl.NumberFormat(locale, {
+    const amount = new BigNumber(value);
+
+    if (amount.isNaN() || !amount.isFinite()) {
+        return '';
+    }
+
+    const wholeNumber = BigInt(amount.toFixed(0, BigNumber.ROUND_DOWN));
```

Comment:
```suggestion
    const roundedDown = amount.toFixed(0, BigNumber.ROUND_DOWN)
    const wholeNumber = BigInt(roundedDown);
```

This can be reused on line 26.

Reply:
4eaafaa841c1e28c7df48e77bb8cac4ac2896b36